tests/helpers/perf_run_helper.rb

```diff
@@ -93,13 +94,42 @@ def perf_teardown
   # @example:
   #   create_perf_archive_root
   #
-  def create_perf_archive_root
+  def create_perf_archive_area
     @gplt_timestamp = Time.now.getutc.to_i
     @archive_root = "#{PERF_RESULTS_DIR}/PERF_#{@gplt_timestamp}"
     FileUtils.mkdir_p @archive_root
+    FileUtils.mkdir_p "#{@archive_root}/hosts"
+    hosts.each do |host|
+      role = get_most_relevant_role(host)
+      role_dir = "#{@archive_root}/hosts/#{role}"
+      FileUtils.mkdir_p("#{role_dir}/#{host.hostname}")
+    end
+  end
+
+  # Pics out the most relevant role from the hosts roles
+  #
+  # @author Randell Pelak
+  #
+  # @param [Object] host The beaker host object
+  #
+  # @return [string] role
+  #
+  # @example:
+  #   get_most_relevant_role(host)
+  #
+  def get_most_relevant_role(host)
+    roles = host[:roles]
+    return "master" if roles.include?("master")
+    return "metric" if roles.include?("metric")
+    return "compiler" if roles.include?("compile_master")
+    return "database" if roles.include?("database")
+    return "loadbalancer" if roles.include?("loadbalancer")
+
+    return "unknown"
```

Comment:
This seems like a weird value to return.  If the given role set for a host is not one of the ones in the list that we care about I would expect that we would return on of the roles actually assigned to the host rather than unknown.  For example, when we get to true XL we will have a 'postgres' role.  It would be helpful to have the host tree reflect that value rather than 'unknown'.

I understand that there is no way to determine a suitable candidate out of an arbitrary list of roles. Perhaps they could be concatenated?

Reply:
Unknown signals that the host isn't one of the currently "relevant" roles.  If I returned the first role on the list when none matched, than it wouldn't be clear to the user that the host doesn't have what is classified as a relevant role (and that likely the code needs updating).   Also if I just grab the first one... it could potentially be different from run to run... order of the roles is not guaranteed. I could change it to none since the host didn't have any of the relevant roles, but I didn't think that looked as good in the directory tree and unknown.

Reply:
My suggestion was to concatenate the roles not to pick the first one from the list.  As a user, how do I know what roles are considered "relevant" by the code paths that this method may be used in?

Reply:
Good point.  How about instead of making a really long directory name (some have like 5 or 6 roles), instead I keep the unknown, and add a file in the host specific dir outlining what the roles were, and a auto generated message about what are currently considered relevant.   kinda like a debug log.  in fact I can call it role_debug.log... or could call it why_unknown.log... thoughts?  @billclaytor any thoughts on this?

Reply:
:-1: on another file.  This information is captured in the beaker configuration.

It sounds like we don't have a great solution to this case.  So, I am fine with using "unknown" for now and ticketing a future spike to try and resolve it.

Reply:
What if there are multiple hosts that wind up with 'unknown'?

Reply:
under the role there is a dir that is the hostname... so they would each get there own.  I hated doing that, but compilers will have multiple, so we needed another dir level anyway.